Update the documentation on installing with Typst (#1941)
jupyter-book/mystmd: docs/creating-pdf-documents.md

## Rendering PDFs with Typst

> [Typst](https://typst.app) is a markup-based typesetting language. Compared to $\LaTeX$, syntax is streamlined and consistent, and compile time is significantly faster.

Author: ```suggestion
[Typst](https://typst.app) is a markup-based typesetting language. It is **significantly faster and simpler than using $\LaTeX$** with results of equal or better quality.
```

> [Typst](https://typst.app) is a markup-based typesetting language. It is **significantly faster and simpler than using $\LaTeX$** with results of equal or better quality.

Reviewer: This removes the context that myst is one of two PDF compilation engines supported by myst; probably still useful context?

Author: good point - I'll add it to the top

(typst:install)=
### How to install Typst

Follow [the Typst installation instructions](https://github.com/typst/typst?tab=readme-ov-file#installation) for several options to install Typst.

> We **strongly recommend using the latest releases of Typst**. If you get a confusing Typst error, a good first step is to upgrade your version of Typst.

Reviewer: Should we specifically warn against installing with npm, since that is such a common thing to do with javascript packages?

> We recommend installing Typst with Mamba, like so:

Reviewer 2: Similar to npm, the conda versions are not maintained by the core team (as far as I can tell). So these will lag behind and/or could be abandoned. My recommendation would be to follow the typst docs directly.

Author: ok will remove that

Reviewer 3: I'm in favour of keeping mamba instructions. Many packages on conda-forge are community maintained, and it makes thinks a lot easier if users are pulling other things from conda. I think we focus on what works well right now. Unless we find that the version on forge is already too old for our required feature set.

Author: Personally I agree with your point about mamba - I included it originally because it's how we recommend installing myst for Python users, and the conda forge version of typst seemed to be updated.

That said I don't have a lot of mental bandwidth for refining this PR so I'd prefer if we just got this to a stage that was safe to merge and then did it. If folks want to debate whether mamba should make it into the docs then I'm happy for somebody to take over the PR.

    output: exports/my-document.pdf
---
```
Finally, build the PDF output with Typst using the following command:

```bash

> myst build --typst

Reviewer 3: For now, I'm going to update this for the single-document workflow despite the points above -- we'll iterate at some point to cover `myst.yml`
```suggestion
myst build article.md --typst
```

```

> You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.

Reviewer: Out of curiosity, is this recommended? `myst build --typst` will probably only work if you have a `myst.yml` file, so isn't that then the place to put the options?

Author: I am not sure - I think a lot of mystmd originated with the assumption that you were working in individual documents / articles, and thus put frontmatter into the `.md` file. The move into the `executablebooks/` org brought another major use-case: multi-page books, and has significantly boosted the importance of `myst.yml`. But I feel like there are still a lot of legacy mentions of article-focused workflows that don't hold anymore.

My intuition is that we should just merge this one for now since I don't really know the answer to your question. I am happy to try and brainstorm an answer to your question w/ you as a follow up.

---
```

> 
> You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.
> The Typst templates use the [MyST templating library](xref:jtex) and support the same configuration as [$\LaTeX$](#render-latex).
> 
> ## Choosing a Template

Author: ````suggestion
Finally, build the PDF output with Typst using the following command:

```bash
myst build --typst
```

You can use [document frontmatter](./frontmatter.md) to control various aspects of your Typst outputs.
The Typst templates use the [MyST templating library](xref:jtex) and support the same configuration as [$\LaTeX$](#render-latex).

## Choosing a Template
````
